NN_module/test_utils.f90

```diff
@@ -1,9 +1,12 @@
 module test_utils
 
-  use :: precision, only: sp, dp
+  ! use :: precision, only: sp, dp
 
   implicit none
 
+  integer, parameter :: sp = 4
```

Comment:
If we not using the precision module, maybe a more telling name (not just two letters) would be better

Reply:
So, this is worth a discussion.
I would prefer to use sp and dp from iso_env_fortran.
However, since CAM is an older code it uses `real(8)`.
So to compile we need to be consisyent - see commit mesage on https://github.com/m2lines/convection-parameterization-in-CAM/pull/44/commits/399612b1f5eb6cdbb5c20765907965f0ffdea33f

Maybe we can discuss this in person?

Reply:
(I was just thinking that `sp` and `dp` are reasonable common Fortran idioms. At least some comment could be added here to explain).

Reply:
After some discussion over this with Kacper I think we perhaps use the iso_fortran_env, include the precision module, and make it the responsibility of the user to set it appropriately.

This will increase portability, and "adding lots of files" is perhaps a weak argument against sensibly architectured and abstracted software.

Reply:
https://github.com/m2lines/convection-parameterization-in-CAM/pull/44/commits/632aa145ba116c8e3053149a4fdfe3eb8f8c2d36 Should have implemented this. It will now be a requirement for those using the scheme to set precision either in the precision file, or set in each file as suits them.